product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java

```diff
@@ -103,38 +111,159 @@ public void testVisibilityOfAPISRestrictedByMultipleRoles() throws Exception {
         createUser(userName, password, new String[]{subscribeRole} , ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
 
         String multipleRoles = subscribeRole + "," + creatorRole;
-        APIRequest apiRequest = new APIRequest(apiName, apiContext, apiVisibility, multipleRoles, VISIBILITY_TYPE, apiVersion, apiResource,
-                tierCollection, new URL(backendEndPoint));
+        APIRequest apiRequest = new APIRequest(apiName, apiContext, apiVisibility, multipleRoles, VISIBILITY_TYPE,
+                apiVersion, apiResource, tierCollection, new URL(backendEndPoint));
 
         validateRoles(multipleRoles);
-        createAPI(apiRequest, multipleRoles);
-        getAPI();
-        publishAPI(apiName, ADMIN_LOGIN_USERNAME);
+        createAPI(apiRequest);
+        getAPI(PUBLISHER_CREATOR_USERNAME);
+        publishAPI(apiName, PUBLISHER_CREATOR_USERNAME);
         loginToStore(userName, password);
         isAPIVisibleInStore(apiName, apiStoreClient);
     }
 
+//    todo need to config a secondary userstore to check roles with spaces
```

Comment:
Shall we replace these commented codes if the todo is done

Reply:
The requirement is not implemented yet